sqlx-core/src/postgres/types/json.rs

```diff
@@ -41,7 +41,7 @@ impl Type<Postgres> for &'_ JsonRawValue {
 
 impl Encode<Postgres> for &'_ JsonRawValue {
     fn encode(&self, buf: &mut PgRawBuffer) {
-        Json(self).encode(buf)
+        (Box::new(Json(self)) as Box<dyn Encode<Postgres>>).encode(buf)
```

Comment:
Oh I see why you did this. `Json(self).encode(buf)` conflicts now.

```rust
<Json<Self> as Encode<Postgres>>::encode(Json(self), buf)
```

^ That should do it without the `Box`